File: 2023/day01/part1.mjs
import { readTextFile } from "../../io.mjs"

const lines = readTextFile("2023/day01/input.txt")

const sum = lines.reduce((sum, line) => {
	const digits = Array.from(line.matchAll(/[0-9]/g))
	if (digits.length === 0) return sum
	const firstDigit = digits[0][0]
	const lastDigit = digits[digits.length - 1][0]

	return sum + parseInt(firstDigit + lastDigit)
}, 0)

console.log(sum)

File: 2023/day01/part1.rs
use std::fs::read_to_string;

fn main() {
	let lines: Vec<String> = read_to_string("input.txt")
		.unwrap()
		.lines()
		.map(String::from)
		.collect();

	let mut sum: u32 = 0;

	for line in lines {
		let digits: Vec<u32> = line
			.chars()
			.filter_map(|c| c.to_digit(10))
			.collect();

		let digit: u32 = (digits[0].to_string() + &digits[digits.len() - 1].to_string())
			.parse()
			.unwrap();
		
		sum += digit;
	}

	println!("{sum}");
}